src/main/java/com/squareup/kotlinpoet/TypeSpec.kt

```diff
@@ -627,7 +628,7 @@ class TypeSpec private constructor(builder: TypeSpec.Builder) {
 
       val isAbstract = ABSTRACT in kind.modifiers || kind !is Kind.Class || !isSimpleClass
       for (funSpec in funSpecs) {
-        require(isAbstract || ABSTRACT !in funSpec.modifiers) {
+        require(isAbstract || ABSTRACT !in funSpec.modifiers || SEALED !in funSpec.modifiers) {
```

Comment:
This is incorrect, you're basically checking that none of the functions inside this `TypeSpec` have `sealed` modifier. Since `sealed` doesn't make sense for functions - the check will always succeed. Instead, you should be treating a `sealed` class as abstract by modifying `val isAbstract` above:

```
val isAbstract = ABSTRACT in kind.modifiers || SEALED in kind.modifiers || kind !is Kind.Class || !isSimpleClass
```

Reply:
doh, you're right.  will correct that.

Reply:
It's odd that the test passed with that wonky check, though.  I'd looked at isAbstract earlier but i clearly got some wires crossed because i was looking at a field on the class instead.